runtime/elem/Value.h

```diff
@@ -96,115 +104,161 @@ namespace js
         Object const& getObject()               const { return std::get<Object>(var); }
         Function const& getFunction()           const { return std::get<Function>(var); }
+        Number const& getNumber()               const { return std::get<Number>(var); }
 
         Array& getArray()                   { return std::get<Array>(var); }
         Float32Array& getFloat32Array()     { return std::get<Float32Array>(var); }
         Object& getObject()                 { return std::get<Object>(var); }
         Function& getFunction()             { return std::get<Function>(var); }
+        Number& getNumber()                 { return std::get<Number>(var); }
 
-        //==============================================================================
-        // Object property access with a default return value
-        template <typename T>
-        T getWithDefault(std::string const& k, T const& v) const
-        {
-            auto o = getObject();
-
-            if (o.count(k) > 0) {
-                return T(o.at(k));
-            }
-
-            return v;
-        }
-
-        // String representation
-        String toString() const
-        {
-            if (isUndefined()) { return "undefined"; }
-            if (isNull()) { return "null"; }
-            if (isBool()) { return String(std::to_string(std::get<Boolean>(var))); }
-            if (isNumber()) { return String(std::to_string(std::get<Number>(var))); }
-            if (isString()) { return std::get<String>(var); }
-            if (isArray())
+         //==============================================================================
+            // Object property access with a default return value
+            template <typename T>
+            T getWithDefault(std::string const& k, T const& v) const
             {
-                auto& a = getArray();
-                std::stringstream ss;
-                ss << "[";
+                auto o = getObject();
 
-                for (size_t i = 0; i < std::min((size_t) 3, a.size()); ++i)
-                    ss << a[i].toString() << ", ";
-
-                if (a.size() > 3)
+                if (o.count(k) > 0)
                 {
-                    ss << "...]";
-                    return ss.str();
+                    return T(o.at(k));
                 }
 
-                auto s = ss.str();
-                return s.substr(0, s.size() - 2) + "]";
+                return v;
             }
-            if (isFloat32Array())
-            {
-                auto& fa = getFloat32Array();
-                std::stringstream ss;
-                ss << "[";
 
-                for (size_t i = 0; i < std::min((size_t) 3, fa.size()); ++i)
-                    ss << std::to_string(fa[i]) << ", ";
 
-                if (fa.size() > 3)
+            // an additional trait to get a vector of strings
+            std::vector<std::string> toStringVector() const
+            {
+                std::vector<std::string> array_of_strings;
```

Comment:
I'm ok to take this method but it seems inconsistent with the `getNumber` style methods in the sense that the other methods throw if you're trying to `get` the wrong value type. In this case it won't throw it will silently return you an empty vec<string>. I'd prefer consistency with the other getters. So perhaps here we skip the isArray() check and just allow getArray() to throw if it's an invalid access?